generators/app/templates/src/StarterKit/Shared/Extensions/ServiceAgentExtensions.cs

```diff
@@ -0,0 +1,149 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Reflection;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using StarterKit.ServiceAgents._base;
+using StarterKit.ServiceAgents._base.Auth;
+using StarterKit.ServiceAgents._base.Helper;
+using StarterKit.ServiceAgents._base.Settings;
+
+namespace StarterKit.Shared.Extensions
+{
+  public static class ServiceAgentExtensions
+  {
+    public static IServiceCollection AddServiceAgents(this IServiceCollection services, IConfigurationSection configSection, Assembly serviceAgentAssembly, Assembly sharedAssembly,
```

Comment:
Deze implementatie van serviceAgents lijkt deels terug te gaan naar de implementatie in de oude ServiceAgent toolbox waarin serviceAgents vanuit Configuratie werden toegevoegd. Bij de meer recenter implementaties van serviceAgents werden die eerder individueel gedefinieerd met gebruik van delegating handlers, ook voor foutafhandeling ipv een generische implementatie te gebruiken. Worden er in deze implementatie nog Delegating handlers gebruikt of verloopt dat via die "customRegistrations"?

Delegating handlers biedt wel een propere manier om per ServiceAgent de specifieke logica te definiëren (bv. wel of niet OAuth2 authorization toevoegen bij call), plus soms andere logica op te zetten voor eenzelfde "type" delegatingHandler bij verschillende ServiceAgents (bv. andere afhandeling van errors omdat serviceAgent ander error-model teruggeeft). Eventueel kunnen we dit stuk even in een online vergadering overleggen.

Zie ook opmerkingen bij  AgentBase. Dit lijkt me teveel toegespitst om enkel interne API's aan te spreken die volledig voldoen aan alle API-requirements.

Reply:
@erikseynaeve, dit heb ik in het verleden inderdaad opgelost via die custom registrations. Nu begrijp ik jou punt wel volledig dat werken via DelegatingHandlers een duidelijkere en properdere oplossing is.

Ik ben deels teruggevallen op de oude ServiceAgent implementatie omdat dit voor de 'meeste' Digipolis projecten waarbij er enkel een koppeling is met services uit de API store een makkelijkere opzet biedt. Zo moet er geen rekening gehouden worden met welke handler al dan niet te gebruiken of aparte options files per service agent. 

Mogelijks is er een gulden middenweg, maar dit bespreken we inderdaad best even in een overleg.